bundle/edu.gemini.qv.plugin/src/main/scala/edu/gemini/qv/plugin/util/SolutionProvider.scala

```diff
@@ -25,9 +27,12 @@ object SolutionProvider {
     Site.GS -> new SolutionProvider(Site.GS)
   )
 
-  def apply(site: Site) = providers(site)
-  def apply(peer: Peer) = providers(peer.site)
-  def apply(ctx: QvContext) = providers(ctx.site)
+  def currentNight(site: Site, currentTime: Long): Option[Night] =
+    SemesterData.current(site, currentTime).nights.find(n => n.dayEnd > currentTime)
```

Comment:
I know it's not really the point of this bug fix, but I thought I'd point it out nevertheless:  I think this lookup could be done in near constant time if we hash the `Night` objects by start time.  There is code in `edu.gemini.skycalc` for this sort of thing.